lib/types/rules/possible-errors.d.ts

```diff
@@ -251,7 +251,23 @@ export interface PossibleErrors extends Linter.RulesRecord {
      * @since 0.4.0
      * @see https://eslint.org/docs/rules/no-extra-boolean-cast
      */
-    "no-extra-boolean-cast": Linter.RuleEntry<[]>;
+    "no-extra-boolean-cast": Linter.RuleEntry<
+        [
+            Partial<{
+                /**
+                 * @since 9.3.0
+                 * @default false
+                 */
+                enforceForInnerExpressions: boolean;
+                /**
+                 * @deprecated
+                 * @since 7.0.0-alpha.2
+                 * @default false
+                 */
+                enforceForLogicalOperands: boolean;
```

Comment:
Only one of the options can be present in the object.

https://github.com/kirkwaiblinger/eslint/blob/a7016a50d88011f279d52b9355a5662e561c414c/lib/rules/no-extra-boolean-cast.js#L33-L54

Reply:
Indeed, I’ve just removed the deprecated option accordingly as the bundled types are a fairly new addition anyway.

Thanks for pointing this out 🙌